libraries/Microsoft.Bot.Builder.BotFramework/BotFrameworkAdapter.cs

```diff
@@ -50,15 +52,34 @@ public BotFrameworkAdapter(string appId, string appPassword) : base()
             }
         }
 
-        public async Task Receive(string authHeader, Activity activity)
+        public async Task Receive(AspNetCore.Http.HttpRequest httpRequest)
```

Comment:
>AspNetCore.Http.HttpRequest  [](start = 34, length = 28)

This is exactly why we didn't do this.  We passed the headers in to remove the hard dependency on AspNetCore.Http.HttpRequest.  By doing this you make it so that this library can't be used on ASP.NET (or other Http frameworks for that matter.)

Reply:
If we wanted to make a .AspNetCore library and a .AspNet library etc. then we could do that, but it increases our surface area.

---
In reply to: [168661056](https://github.com/Microsoft/botbuilder-dotnet/pull/136#discussion_r168661056) [](ancestors = 168661056)

Reply:
Yup I wondered about that. However if it's a .Net Standard lib it *should* be consumable by .Net 4.6.1+ w/o issue. I'm working on proofing this and putting in a branch as we speak.

Reply:
If you've got a moment, please take a look at PR #145, as it may accomplish what you're looking to do here.

Reply:
unfortunately it doesn't. it still has:
```csharp
public async Task ProcessActivty(string authHeader, Activity activity, Func<IBotContext,Task> callback)
```
in it which requires me as the consumer to know exactly what/where the auth header is and pass it in. It also ties us to only ever having one header come in - what if we need another one in the future? Breaking change for all bots. Those two points are what I'd like to see alleviated.